__test__/server/api/campaign.test.js

```diff
@@ -778,6 +781,140 @@ describe("Reassignments", async () => {
     expect(texterCampaignDataResults2.data.assignment.allContactsCount).toEqual(
       29
     );
-    console.timeEnd('fullfunction');
-  }, 10000);  // long test can exceed default 5seconds
+    console.timeEnd("fullfunction");
+  }, 10000); // long test can exceed default 5seconds
+});
+
+describe("Bulk Send", async () => {
+  const OLD_ENV = process.env;
+
+  beforeEach(async () => {
+    jest.resetModules(); // this is important - it clears the cache
+    process.env = {
+      ...OLD_ENV
+    };
+  });
+
+  afterEach(async () => {
+    process.env = OLD_ENV;
+  });
+
+  const testBulkSend = async (params, expectedSentCount, resultTestFunction) => {
+    process.env.ALLOW_SEND_ALL = params.allowSendAll;
+    process.env.NOT_IN_USA = params.notInUsa;
+    process.env.BULK_SEND_CHUNK_SIZE = params.bulkSendChunkSize;
+
+    testCampaign.use_dynamic_assignment = true;
+    await createScript(testAdminUser, testCampaign);
+    await startCampaign(testAdminUser, testCampaign);
+    let texterCampaignDataResults = await runComponentGql(
+      TexterTodoQuery, {
+        contactsFilter: {
+          messageStatus: "needsMessage",
+          isOptedOut: false,
+          validTimezone: true
+        },
+        assignmentId
+      },
+      testTexterUser
+    );
+
+    // TEXTER 1 (NUMBER_OF_CONTACTS needsMessage)
+    expect(texterCampaignDataResults.data.assignment.contacts.length).toEqual(
+      NUMBER_OF_CONTACTS
+    );
+    expect(texterCampaignDataResults.data.assignment.allContactsCount).toEqual(
+      NUMBER_OF_CONTACTS
+    );
+
+    // send some texts
+    const bulkSendResult = await bulkSendMessages(assignmentId, testTexterUser);
+    console.log(bulkSendResult);
+    resultTestFunction(bulkSendResult);
+
+    // TEXTER 1 (95 needsMessage, 5 needsResponse)
+    texterCampaignDataResults = await runComponentGql(
+      TexterTodoQuery, {
+        contactsFilter: {
+          messageStatus: "needsMessage",
+          isOptedOut: false,
+          validTimezone: true
+        },
+        assignmentId
+      },
+      testTexterUser
+    );
+
+    expect(texterCampaignDataResults.data.assignment.contacts.length).toEqual(
+      NUMBER_OF_CONTACTS - expectedSentCount
+    );
+    expect(texterCampaignDataResults.data.assignment.allContactsCount).toEqual(
+      NUMBER_OF_CONTACTS
+    );
+  };
+
+  const expectErrorBulkSending = (result) => {
+    expect(result.errors[0]).toBeDefined();
```

Comment:
It's fine to make this ambiguous if it's not consistent.  Can you add a comment on what it outputs?